primitives/src/header.rs

```diff
@@ -145,13 +146,20 @@ where
 	Hash::Output: Decode,
 {
 	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		let parent_hash = Decode::decode(input)?;
```

Comment:
Curios about extracting inlined code to vars. Is reasoning to ease debugging?

Reply:
Yes, just easy debugging.